tracking/translations_parser/parser.py

```diff
@@ -0,0 +1,288 @@
+import logging
+import re
+from collections import defaultdict
+from collections.abc import Iterable, Sequence
+from datetime import datetime
+from typing import Callable
+
+import yaml
+
+from translations_parser.data import Metric, TrainingEpoch, TrainingLog, ValidationEpoch
+from translations_parser.publishers import Publisher
+
+logging.basicConfig(
+    level=logging.INFO,
+    format="[%(levelname)s] %(message)s",
+)
+logger = logging.getLogger(__name__)
+
+HEADER_RE = re.compile(r"(?<=\[)(?P<value>.+?)\] ")
+VALIDATION_RE = re.compile(
+    r"Ep\.[ :]+(?P<ep>\d+)[ :]+Up\.[ :]+(?P<up>\d+)[ :]+(?P<key>[\w-]+)[ :]+(?P<value>[\d\.]+)"
+)
+TRAINING_RE = re.compile(
+    r"Ep\.[ :]+(?P<epoch>\d+)[ :]+"
+    r"Up\.[ :]+(?P<up>\d+)[ :]+"
+    r"Sen\.[ :]+(?P<sen>[\d,]+)[ :]+"
+    r"Cost[ :]+(?P<cost>[\d.]+)[ :]+"
+    r"Time[ :]+(?P<time>[\d\.]+)s[ :]+"
+    r"(?P<rate>[\d\.]+) words\/s[ :]+"
+    r"gNorm[ :]+(?P<gnorm>[\d\.]+)"
+)
+# Expected version of Marian for a clean parsing
+MARIAN_MAJOR, MARIAN_MINOR = 1, 10
```

Comment:
I think the parsing logic could remain the same, and different regular expressions would be used depending on the version (default to latest in case Marian context is missing). Thank you for the linking, so this issue is visible.